third_party/move/move-compiler/src/expansion/ast.rs

```diff
@@ -548,6 +548,44 @@ pub enum Exp_ {
 }
 pub type Exp = Spanned<Exp_>;
 
+impl Exp_ {
+    /// Get all names from an expression
+    /// only recursively check on expression types supported in constant definition.
+    pub fn get_names_for_const_exp(&self) -> HashSet<Name> {
```

Comment:
Could we please use a `BTreeSet` instead? Looks like we iterate through this hashset later, and such iteration order is not deterministic.

Reply:
```suggestion
    pub fn get_names_from_const_exp(&self) -> HashSet<Name> {
```

Reply:
done